### PR TITLE
Expose config max-round-blocks-to-import

### LINKED ISSUE #9320: Parity Full Node Occasionally Skips Blocks
I'm running:
 - **Which Parity version?**: v1.11.8
 - **Which operating system?**: Docker parity/parity:v1.11.8
 - **How installed?**: Docker
 - **Are you fully synchronized?**: yes
 - **Which network are you connected to?**: ropsten
 - **Did you try to restart the node?**: yes

We're currently running a full parity node on Ropsten and seeing some notifications for new blocks being dropped. We're connecting to a Parity node with pubsub via websocket connection and can verify that our node did not receive the skipped blocks.

Parity output (missed block 3797581):
```
2018-08-08 18:03:27 UTC Imported #3797579 0x981e…fc53 (75 txs, 5.65 Mgas, 725.03 ms, 21.79 KiB)
2018-08-08 18:03:46 UTC   25/25 peers      5 MiB chain   72 MiB db  0 bytes queue   90 KiB sync  RPC:  1 conn,  0 req/s,  76 µs
2018-08-08 18:03:49 UTC Imported #3797580 0x3680…e5c1 (68 txs, 3.57 Mgas, 873.04 ms, 14.52 KiB)
2018-08-08 18:03:50 UTC Imported #3797582 0xb9d8…cf6e (31 txs, 3.22 Mgas, 223.74 ms, 13.02 KiB)
2018-08-08 18:04:11 UTC Imported #3797583 0xc2ee…ba30 (4 txs, 2.18 Mgas, 28.82 ms, 8.65 KiB)
2018-08-08 18:04:21 UTC   25/25 peers      5 MiB chain   72 MiB db  0 bytes queue   90 KiB sync  RPC:  1 conn,  0 req/s,  76 µs
```

Our application output (verifies that it also didn't see block 3797581 on the Parity node):
```
{"level":"debug","ts":1533751407.759711,"caller":"services/head_tracker.go:221","msg":"Received header #3797579 (0x39f24b) with hash 0x2a3183b0a8f6c062f8577272a6f0ec4c699c20b5ea7a8b7721df740615ea6512","hash":"0x2a3183b0a8f6c062f8577272a6f0ec4c699c20b5ea7a8b7721df740615ea6512"}
{"level":"debug","ts":1533751429.2156775,"caller":"services/head_tracker.go:221","msg":"Received header #3797580 (0x39f24c) with hash 0x921a45c0763d6be52f4f97c8d98edc2ca3ef5b4e2712a9cf99ccb63da0a9df8d","hash":"0x921a45c0763d6be52f4f97c8d98edc2ca3ef5b4e2712a9cf99ccb63da0a9df8d"}
{"level":"debug","ts":1533751430.88954,"caller":"services/head_tracker.go:221","msg":"Received header #3797582 (0x39f24e) with hash 0xabb65b548e4462108ace90a796639511e13ce86c11802056979988ca3ce8e905","hash":"0xabb65b548e4462108ace90a796639511e13ce86c11802056979988ca3ce8e905"}
{"level":"debug","ts":1533751451.3204648,"caller":"services/head_tracker.go:221","msg":"Received header #3797583 (0x39f24f) with hash 0xf5ae32cc6aa0a96f343070ff8c8212250bf088284c461440aff6c21f82d3d4e5","hash":"0xf5ae32cc6aa0a96f343070ff8c8212250bf088284c461440aff6c21f82d3d4e5"}
```

Sometimes, when this happens, there's a line like `+ another 2 block(s) containing 137 tx(s)` that "catches up" the missed blocks, but this does not happen every time. Can anything be done when Parity skips blocks and does not account for the skipped blocks?

### PR DESCRIPTION
**Pull request by sorpaas**
Fixes #9320.

parity-ethereum imports blocks in rounds. If at the end of any round, the queue is not empty, we consider it to be "importing" and won't notify pubsub. On large re-orgs (10+ blocks), this is possible.

* Change default `max_round_blocks_to_import` number from `4` to `12`.
* Make this configurable via `--max-round-blocks-to-import [S]`. With unstable network conditions, it is advised to increase the number. This shouldn't have any noticeable performance impact unless the number is set to really large.

### CODE REVIEW

File: parity/blockchain.rs [hunk 1]
```diff
@@ -98,6 +98,7 @@ pub struct ImportBlockchain {
 	pub with_color: bool,
 	pub verifier_settings: VerifierSettings,
 	pub light: bool,
+	pub max_round_blocks_to_import: usize,
```
Comment (andresilva) [inline parity/blockchain.rs]: We probably don't need to add this field here and just use the default (12). Same below for `Export*` structs.

File: parity/snapshot.rs [hunk 2]
```diff
@@ -62,6 +62,7 @@ pub struct SnapshotCommand {
 	pub file_path: Option<String>,
 	pub kind: Kind,
 	pub block_at: BlockId,
+	pub max_round_blocks_to_import: usize,
```
Comment (andresilva) [inline parity/snapshot.rs]: Same as above, I think we only need to add this for the `RunCmd`.

File: parity/blockchain.rs (hunk 1 above)
Comment (andresilva) [reply parity/blockchain.rs]: It is a valid option indeed, I just assumed it wouldn't make much of a difference without networking. I'm OK with with keeping it configurable.

**Comment by ngotchac** [comment]
Did you experience any performance improvements/decreases defaulting from 4 to 12?

**Comment by sorpaas** [comment]
@ngotchac No. Not sure how to benchmark this.

**Comment by ngotchac** [comment]
Nothing really precise here, but just looking empirically how fast can you sync to block #N I guess? I think it won't change much though.

**Comment by 5chdn** [comment]
Are you still working on this?

**Comment by sorpaas** [comment]
@5chdn Fixed! Didn't realize there were merge conflicts!

**Comment by 5chdn** [comment]
Could you take a look @ngotchac @tomusdrw :)

File: parity/cli/mod.rs [hunk 3]
```diff
@@ -785,6 +785,10 @@ usage! {
 			"--stratum-secret=[STRING]",
 			"Secret for authorizing Stratum server for peers.",
 
+			ARG arg_max_round_blocks_to_import: (usize) = 12usize, or |c: &Config| c.mining.as_ref()?.max_round_blocks_to_import.clone(),
```
Comment (tomusdrw) [inline parity/cli/mod.rs]: Doesn't it fit more into `Performance` section?

**Comment by 5chdn** [comment]
please rebase on master :see_no_evil:

**Pull request merged**